File: API/Enitites/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Enitites
{
    public class Product
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public long Price { get; set; }
        public string Type { get; set; }
        public string Brand { get; set; }
        public int Quantity { get; set; }
        public string PictureUrl { get; set; }
        public long QuantityInStock { get; set; }
    
    public Basket b{get;set;}
    }
}

File: client/src/app/model/Product.ts


export interface Product {
    id: number;
    name: string;
    description: string;
    price: number;
    type?: string;
    brand: string;
    quantity: number;
    pictureUrl: string;
    quantityInStock?: number;
}

export interface ProductParams {
    orderBy: string;
    searchTerm?: string;
    types?: string[];
    brands?: string[];
    pageNumber: number;
    pageSize: number
}


